### PR TITLE
Use request.URL.Host instead when request.Host is empty

### PR DESCRIPTION
**Pull request by StoneMoe**
As we discussed in Slack, this will add support for HTTP requests with empty `request.Host`, such as elasticsearch client, which set request URL after a path-only request creation.

Ref: https://the-asf.slack.com/archives/CHQ1L4T54/p1734924623039789

### CODE REVIEW

File: plugins/http/client_intercepter.go [hunk 1]
```diff
@@ -30,12 +30,18 @@ type ClientInterceptor struct {
 
 func (h *ClientInterceptor) BeforeInvoke(invocation operator.Invocation) error {
 	request := invocation.Args()[0].(*http.Request)
-	s, err := tracing.CreateExitSpan(fmt.Sprintf("%s:%s", request.Method, request.URL.Path), request.Host, func(headerKey, headerValue string) error {
+	host := request.Host
+	if host == "" {
+		if request.URL != nil {
```
Comment (mrproliu) [inline plugins/http/client_intercepter.go:35]:
```suggestion
	if host == "" && request.URL != nil {
```
I think it can be simplified to a single line.

**Comment by mrproliu** [comment]
Please update the [CHANGES.md](https://github.com/apache/skywalking-go/blob/main/CHANGES.md) file.

**Comment by mrproliu** [comment]
Reference to the slack link: https://the-asf.slack.com/archives/CHQ1L4T54/p1734924623039789

**Comment by wu-sheng** [comment]
image

Binary is not allowed. Please remove this unexpected file.

**Comment by mrproliu** [comment]
@StoneMoe, please fix the CI failure.

**Comment by StoneMoe** [comment]
> @StoneMoe, please fix the CI failure.

It says:
```
Error: instrument.go:38:9: string `http` has 2 occurrences, make it a constant (goconst)
	return "http"
	       ^
```

Maybe we should fix it by incresing goconst `min-occurrences` options? @mrproliu

**Comment by mrproliu** [comment]
Can you help to check why there say `2 occurrences`, If so, we should add a new const here.

**Comment by StoneMoe** [comment]
> Can you help to check why there say `2 occurrences`, If so, we should add a new const here.

In test file there is a new "http" string I added.

Considering change goconst option `ignore-tests` to `true` instead I think.

**Review by mrproliu — APPROVED** [review]

**Pull request merged**